sale_coupon_auto_refresh/models/sale_order.py

```diff
@@ -49,19 +49,25 @@ class SaleOrderLine(models.Model):
 
     @api.model_create_multi
     def create(self, vals_list):
-        lines = super().create(vals_list)
+        lines = super(
+            SaleOrderLine, self.with_context(skip_auto_refresh_coupons=True)
+        ).create(vals_list)
         lines.mapped("order_id")._auto_refresh_coupons()
```

Comment:
Another thing to improve the `_auto_refresh_coupons` method would be to disable it whenever the context has the website_id on it, as there's already a `website_sale_coupon` module triggering the change on every `_cart_update`